app/src/main/java/com/example/sharingang/UserProfileViewModel.kt

```diff
@@ -1,20 +1,35 @@
 package com.example.sharingang
 
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.Transformations
-import androidx.lifecycle.ViewModel
+import androidx.lifecycle.*
+import com.example.sharingang.items.Item
+import com.example.sharingang.items.ItemRepository
+import com.example.sharingang.items.ItemsViewModel
+import com.example.sharingang.users.CurrentUserProvider
 import com.example.sharingang.users.User
 import com.example.sharingang.users.UserRepository
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class UserProfileViewModel @Inject constructor(
+    currentUserProvider: CurrentUserProvider,
+    private val itemRepository: ItemRepository,
     private val userRepository: UserRepository
 ) : ViewModel() {
 
-    private val _userId = MutableLiveData<String?>()
+    private val _userId = MutableLiveData<String?>(currentUserProvider.getCurrentUserId())
```

Comment:
When we go to the user profile, it's not necessarily only the current user we'll see. We can also see other users and that's why we use the setUser method to change it.